canvas_modules/common-canvas/src/toolbar/toolbar.jsx

```diff
@@ -422,12 +422,27 @@ class Toolbar extends React.Component {
 
 		if (actionObj) {
 			if (actionObj.divider) {
-				jsx = (
-					<ToolbarDividerItem
-						key={"toolbar-item-key-" + i}
-						isInMenu={false}
-					/>
-				);
+				const isRightBar = refs === this.rightItemRefs;
+				if (!isRightBar) {
+					// No need to add overflow icon for the right toolbar divider.
+					jsx = (
+						<>
+							{this.generateOverflowItem(i, actionObj.action)}
+							<ToolbarDividerItem
+								key={"toolbar-item-key-" + i}
+								isInMenu={false}
+							/>
+						</>
+					);
+				} else {
+					// render only the divider no need to render the overflow icon.
+					jsx = (
+						<ToolbarDividerItem
+							key={"toolbar-item-key-" + i}
+							isInMenu={false}
+						/>
+					);
+				}
```

Comment:
Hi @Jerinjk14 -- I think you've got the right approach for fixing this. However, did you consider removing `!actionObj.divider && ` from this code as a fix:
 
```
generateToolbarItems(toolbarActions, withOverflowItem, refs) {
		const newItems = [];

		for (let i = 0; i < toolbarActions.length; i++) {
			const actionObj = toolbarActions[i];
			if (actionObj) {
				if (!actionObj.divider && withOverflowItem) {
					newItems.push(this.generateOverflowItem(i, actionObj.action));
				}
				newItems.push(this.generateToolbarItem(actionObj, i, refs));
			}
		}
		return newItems;
	}
```